crates/bevy_ecs/src/storage/table.rs

```diff
@@ -32,6 +32,37 @@ impl TableId {
     }
 }
 
+/// A opaque newtype for rows in [`Table`]s. Specifies a single row in a specific table.
+///
+/// Values of this type are retreivable from [`Archetype::entity_table_row`] and can be
+/// used alongside [`Archetype::table_id`] to fetch the exact table and row where an
+/// [`Entity`]'s
+///
+/// Values of this type are only valid so long as entities have not moved around.
+/// Adding and removing components from an entity, or despawning it will invalidate
+/// potentially any table row in the table the entity was previously stored in. Users
```

Comment:
Could spawning an entity invalidate this or does spawning always add a row at the end of a table?

Reply:
Spawning always appends to the table, it's only on table moves where the table row is invalidated for that entity and others in the same table.